internal/services/api_router.go

```diff
@@ -72,3 +73,12 @@ func ApiRouter(deps *ApiServices) http.Handler {
 		fmt.Fprintln(w, "expected grpc request")
 	})
 }
+
+// GRPC events have the nature of DEBUG logs but are logged with INFO level. To clean up the log stream starting from INFO log level we only log WARN events.
+func grpcLoggerWith(ctx context.Context) *logrus.Entry {
+	if (logrus.GetLevel() == logrus.TraceLevel || logrus.GetLevel() == logrus.DebugLevel) {
+		return traces.Logger(ctx)
+	} else {
+		return traces.WarnLogger(ctx)
```

Comment:
This mean we are also "downgrading" error/fatal/panic to warn. Is that intended?

Maybe, we can only set it to warn if the log level is info, like:

(attention: pseudo code)
```go
	if logrus.GetLevel() == logrus.InfoLevel {
		return traces.WarnLogger(ctx)
	} else {
		return traces.Logger(ctx)
    }
```

Reply:
No, the `WarnLogger` does not affect log items with higher level; it swallows items with level lower than `WARN`, but still shows items with levels `WARN`, `ERROR`, `FATAL`, etc.

This can demonstrated by setting the application setting to `loglevel: info` and producing an error while creating a device (e.g. by submitting an empty device name):
```
INFO[0000]main.go:94 +++ wg-access-server development (-)         
INFO[0000]main.go:141 Starting WireGuard on :51820                 
INFO[0000]main.go:157 WireGuard VPN network is 10.10.10.1/24       
INFO[0000]contracts.go:79 Storing data in SQL backend at sqlite3://db.sqlite3/wireguard-db.sqlite3 
INFO[0000]server.go:95 Starting DNS server on 10.10.10.1:53/udp, 10.10.10.1:53/tcp with upstreams: 10.100.10.8 
INFO[0000]devices.go:67 Start collecting device metadata             
INFO[0000]website_router.go:29 Missing ./website/build - will reverse proxy to website dev server 
INFO[0000]main.go:280 Web UI listening on :8000                    
INFO[0496]session.go:46 Creating web session with provider 'simple' for user 'admin' (remote address: [::1]:43894) 
ERRO[0504] Device name must not be empty.                grpc.method=AddDevice grpc.service=proto.Devices grpc.start_time="2023-09-11T13:37:58+02:00" span.kind=server system=grpc trace.id=34c49e1e-9370-4fef-88fc-1f9049c2405d
ERRO[0504] finished unary call with code Internal        error="rpc error: code = Internal desc = Device name must not be empty." grpc.code=Internal grpc.method=AddDevice grpc.service=proto.Devices grpc.start_time="2023-09-11T13:37:58+02:00" grpc.time_ms=0.039 span.kind=server system=grpc trace.id=34c49e1e-9370-4fef-88fc-1f9049c2405d

```

Reply:
Maybe I am missing something, but I meant the other way around: If we set the gobal log level to error/fatal/panic, grpc logs on the warn log level will still be shown.

Reply:
Ah, you are certainly right! :see_no_evil: